birdyboard.py

```diff
@@ -0,0 +1,505 @@
+        elif next_step == "5":  # new private thread.
+            self.public_or_private = "private"
+            self.new_chirp_menu()
+        elif next_step == "x":  # exit.
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.logged_in_menu_next_step()
+
+# ###############################
+# ######## CREATE A USER ########
+# ###############################
+
+    def create_a_user_menu(self):
+        self.deserialize_users()
+        user_name = input("user name: ")
+        if user_name == "b":
+            print("going back")
+            print(self.unlogged_in_menu)
+            self.unlogged_in_menu_next_step()
+        elif user_name == "x":
+            print("goodbye.")
+            exit()
+        else:
+            for user in self.users:
+                if user_name == user["user_name"]:
+                    print("User name already taken!")
+                    self.what_if_user_name_is_taken()
+            password = input("password: ")
+            self.add_new_user(user_name, password)
+            print("logging in " + user_name + ".")
+            print(self.logged_in_menu)
+            self.logged_in_menu_next_step()
+
+    def what_if_user_name_is_taken(self):
+        print("'b' to go back.\n'x' to exit.\n1. Choose from a list of created users.\n2. Try creating a user with a different name.")
+        next_step = input(">> ")
+        if next_step == "1":
+            self.deserialize_users()
+            self.view_users_menu()
+            self.users_menu_next_step()
+        elif next_step == "2":
+            self.create_a_user_menu()
+        elif next_step == "b":
+            print("going back.")
+            print(self.unlogged_in_menu)
+            self.unlogged_in_menu_next_step()
+        elif next_step == "x":
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.what_if_user_name_is_taken()
+
+    def add_new_user(self, user_name, password):
+        self.deserialize_users()
+        self.users.append({"user_name": user_name, "password": password})
+        self.serialize_users()
+        self.user_name = user_name
+
+
+# ############################
+# ######## VIEW USERS ########
+# ############################
+# I am fully aware that this is a completely insecure way to handle a password system.
+
+    def view_users_menu(self):
+        """
+        prints a list of users (after deserialization) that are currently created on the app.
+        Arguments: None
+        """
+        [print(str(self.users.index(user) + 1) + ". " + user["user_name"]) for user in self.users]
+
+    def users_menu_next_step(self):
+        """
+        prints after 'view users menu'. creates input for which existing user is being selected and error handles.
+        arguments: None
+        """
+        next_step = input("'b' to go back.\n'x' to exit.\nChoose a user to log into.\n>> ")
```

Comment:
Logging into a user sounds weird.  Log in _as_ makes more sense in this instance.